### DISCUSSION TITLE
Native and Cooja compiler not the same

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2246] [General]
I noticed that compilations errors caught when compiling with `TARGET=native` will not be reported when the code is compiler by Cooja. Should they be the same (or run with the same parameters)?

**Comment by alexrayne** [comment]
No, their not the same - cooja is a platform on java natve mode environment. That environment armed with specific interfaces - device drivers simulators. 
Native - loks as a generin aplication on some OS. for that OS you need provide drivers on contki devices.

**Reply by salmeghlef** [reply]
Nice information Alexrayne, for the native, you said "for that OS you need to provide drivers on Contiki devices",, please tell us how to do so?

**Comment by pjonsson** [comment]
The build system was unified in v4.8 (and beyond), so the same compiler should be used. Cooja and native are separate targets though, so the target-specific code is different.

This isn't a bug but the information might be useful for someone else, so I think this should be moved to the discussion forum. I don't think the reporter has permission to move their own tickets to the forum though, so let's hope someone else moves this for us.
